rs/messaging/src/state_machine.rs

```diff
@@ -114,6 +118,22 @@ impl StateMachine for StateMachineImpl {
         self.metrics
             .timed_out_messages_total
             .inc_by(timed_out_messages as u64);
+
+        // Time out expired callbacks.
+        let (timed_out_callbacks, errors) = state.time_out_callbacks();
+        self.metrics
+            .timed_out_callbacks_total
+            .inc_by(timed_out_callbacks as u64);
+        for error in errors {
+            // Critical error, responses should always be inducted successfully.
+            error!(
+                self.log,
+                "{}: Inducting deadline expired response failed: {}",
+                CRITICAL_ERROR_INDUCT_RESPONSE_FAILED,
+                error
+            );
```

Comment:
I think bumping the critical error counter is still missing here, no?

Reply:
Oh, seems there was a race condition between you pushing this commit and me reviewing. Anyways, thanks for fixing.